src/StreetNameRegistry/Municipality/MunicipalityStreetName.cs

```diff
@@ -97,6 +98,47 @@ public void CorrectRetirement(Action<Names, HomonymAdditions, PersistentLocalId>
             Apply(new StreetNameWasCorrectedFromRetiredToCurrent(_municipalityId, PersistentLocalId));
         }
 
+        public void CorrectHomonymAdditions(
+            HomonymAdditions homonymAdditions,
+            Action<Names, HomonymAdditions, PersistentLocalId> guardUniqueActiveStreetNameNames)
+        {
+            GuardStreetNameStatus(StreetNameStatus.Proposed, StreetNameStatus.Current);
+
+            foreach (var item in homonymAdditions)
+            {
+                if (!HomonymAdditions.HasLanguage(item.Language))
+                {
+                    throw new CannotCorrectNonExistentHomonymAdditionException(item.Language.ToString());
+                }
+            }
+
+            guardUniqueActiveStreetNameNames(Names, homonymAdditions, PersistentLocalId);
+
+            var changedHomonymAdditions = homonymAdditions.Except(HomonymAdditions).ToList();
+
+            if (changedHomonymAdditions.Any())
+            {
+                Apply(new StreetNameHomonymAdditionsWereCorrected(_municipalityId, PersistentLocalId, changedHomonymAdditions));
+            }
+        }
+
+        public void RemoveHomonymAdditions(
+            List<Language> languages,
+            Action<Names, HomonymAdditions, PersistentLocalId> guardUniqueActiveStreetNameNames)
```

Comment:
new lines

Reply:
It is one Action: Action<Names, HomonymAdditions, PersistentLocalId> guardUniqueActiveStreetNameNames)